types/got/index.d.ts

```diff
@@ -98,6 +87,36 @@ declare namespace got {
         (url: GotUrl, options: GotBodyOptions<null>): GotPromise<Buffer>;
     }
 
+    interface GotJSONFn {
+        (url: GotUrl): GotPromise<any>;
+        (url: GotUrl, options: GotJSONOptions): GotPromise<any>;
+        (url: GotUrl, options: GotFormOptions<string>): GotPromise<any>;
+        (url: GotUrl, options: GotBodyOptions<string>): GotPromise<any>;
+    }
+
+    type GotInstance<T = GotFn> = T &
+        Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', T> &
+        {
+            stream: GotStreamFn & Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', GotStreamFn>;
+            extend: GotExtend;
+            RequestError: typeof RequestError;
+            ReadError: typeof ReadError;
+            ParseError: typeof ParseError;
+            HTTPError: typeof HTTPError;
+            MaxRedirectsError: typeof MaxRedirectsError;
+            UnsupportedProtocolError: typeof UnsupportedProtocolError;
+            CancelError: typeof CancelError;
+            TimeoutError: typeof TimeoutError;
+        };
+
+    type DefaultGotInstance = GotInstance;
+    type JSONGotInstance = GotInstance<GotJSONFn>;
+
+    interface GotExtend {
+        (options: GotJSONOptions): JSONGotInstance;
+        (options: GotFormOptions<string> | GotBodyOptions<string> | GotFormOptions<null> | GotBodyOptions<null>): DefaultGotInstance;
```

Comment:
I think that `GotExtend` should repeat overloads for all special cases from `GotFn`, not only for json.